### PR TITLE
Saved poa/appellant emails to database

### PR DESCRIPTION
**Pull request by sahalliburton**
Resolves [CASEFLOW-2025](https://vajira.max.gov/browse/CASEFLOW-2025)

**PR Stack**

[Part 1](https://github.com/department-of-veterans-affairs/caseflow/pull/16616)
Part 2
[Part 3](https://github.com/department-of-veterans-affairs/caseflow/pull/16619)

### Description

Modifies the Schedule Veteran form to include the representative and appellant email fields for all hearing request types.

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] When entered for Video and Central hearings, email addresses are saved to the database

Hearing Type | Help text
-- | --
Virtual | When you schedule the hearing, the Veteran, POA, and Judge will receive an email with connection information for the virtual hearing.
Video, Central | The Veteran and POA will receive email reminders 7 and 3 days before the hearing.Caseflow won’t send notifications immediately after scheduling.
  - [ ] If the appeal does not have a POA, leave ", POA, " or " and POA" out of the help text

### Testing Plan

**Before All**
1.  Open a rails console and enable the feature flag `FeatureToggle.enable!(:collect_video_and_central_emails)`

**Video**
1. Shadow user `BVASYELLOW`: http://localhost:3000/test/users
2. Navigate to the hearing schedule: http://localhost:3000/hearings/schedule
3. Click the “Schedule Veterans” button at the top right
4. Select the St. Petersburg, FL RO from the dropdown
5. Click the “AMA Veterans Waiting” tab and click the name of one of the Veterans
6. On the Case Details page, click the “Schedule Veteran” option from the task list
7. Leave the Hearing Type as "Video"; fill in the hearing location, date and time and then click the "Schedule" button
8. Open a console and locate the hearing that was created: `h = Appeal.find_by(uuid: "[the uuid from the case details page URL]").hearings.last`
9. Ensure there were no email recipients created: `hearing.email_recipients.count == 0`
10. Repeat steps 4-7, but this time input an appellant email before clicking schedule
11. Repeat steps 8-9 and ensure that this time there is an appellant email created
10. Repeat steps 4-7, but this time input an representative email before clicking schedule
11. Repeat steps 8-9 and ensure that this time there is an representative email created
10. Repeat steps 4-7, but this time input both a representative email and an appellant email before clicking schedule
11. Repeat steps 8-9 and ensure that this time there are both a representative email and an appellant email created

*Virtual**
1. Shadow user `BVASYELLOW`: http://localhost:3000/test/users
2. Navigate to the hearing schedule: http://localhost:3000/hearings/schedule
3. Click the “Schedule Veterans” button at the top right
4. Select the St. Petersburg, FL RO from the dropdown
5. Click the “AMA Veterans Waiting” tab and click the name of one of the Veterans
6. On the Case Details page, click the “Schedule Veteran” option from the task list
7. Change the request type to "Virtual"
7. Fill in the hearing location, date and time as well as the appellant email and timezone and then click the "Schedule" button
8. Reload the Case Details page, scroll down to the "View Hearing Details" link that was just created, and click it to go to the hearing details page
9. Ensure that the emails show up in the sent email events table
10. Repeat steps 4-8 but this time leave the appellant email blank and ensure that you receive an error message

### CODE REVIEW

File: client/app/hearings/components/ScheduleVeteran.jsx [hunk 1]
```diff
@@ -94,7 +93,8 @@ export const ScheduleVeteran = ({
   // Create a hearing object for the form
   const hearing = {
     ...assignHearingForm,
-    regionalOffice: assignHearingForm?.regionalOffice?.key,
+    requestType: assignHearingForm?.requestType || appeal?.readableHearingRequestType,
```
Comment (tomas-nava) [inline client/app/hearings/components/ScheduleVeteran.jsx:96]: how & why will this requestType differ from the one defined on line 83?

File: client/app/hearings/components/ScheduleVeteran.jsx [hunk 2]
```diff
@@ -94,7 +93,8 @@ export const ScheduleVeteran = ({
...
+    regionalOffice: assignHearingForm?.regionalOffice?.key || initialRegionalOffice,
```
Comment (tomas-nava) [inline client/app/hearings/components/ScheduleVeteran.jsx:97]: Does this fix a bug? Were we leaving regionalOffice undefined in some cases?

**Review by tomas-nava — CHANGES_REQUESTED** [review]
I left a few suggestions & questions;

also, if I do the following:
- schedule a video hearing with appellant & representative tzs & emails
- go to the hearing's details page
- convert to virtual

... on the "Convert to Virtual Hearing" page, the appellant/rep timezones & emails do not match the email recipients I just created.

File: client/app/hearings/components/ScheduleVeteranForm.jsx [hunk 3]
```diff
@@ -68,7 +68,7 @@ export const ScheduleVeteranForm = ({
     }
 
     // The default is video hearing if the appeal isn't associated with an RO.
-    return appeal?.readableOriginalHearingRequestType ?? VIDEO_HEARING_LABEL;
+    return appeal?.readableHearingRequestType ?? VIDEO_HEARING_LABEL;
```
[the remark this answers is not in the document]
Comment (tomas-nava) [reply client/app/hearings/components/ScheduleVeteranForm.jsx:71]: This method is used to populate the hearing type dropdown and in certain cases (specifically a hearing converted to Central, then being converted to Virtual) was not populating with the correct values.

File: client/app/hearings/components/VirtualHearings/RepresentativeSection.jsx [hunk 4]
```diff
@@ -39,11 +40,11 @@ export const RepresentativeSection = ({
     ) : (
       <ReadOnly text={`The ${appellantTitle} does not have a representative recorded in VBMS`} />
     )}
-    {showTimezoneField && (schedulingToVirtual || userCanCollectVideoCentralEmails) && (
+    {showTimezoneField && (schedulingToVirtual || userCanCollectVideoCentralEmails || virtual) && (
       <div className={classNames('usa-grid', { [marginTop(30)]: true })}>
         <div className={classNames(fullWidth ? 'usa-width-one-whole' : 'usa-width-one-half')}>
           <Timezone
-            optional
+            optional={!virtualHearing?.representativeEmail}
```
[the remark this answers is not in the document]
Comment (tomas-nava) [reply client/app/hearings/components/VirtualHearings/RepresentativeSection.jsx:46]: This is just used to toggle the rep timezone field as optional/required based on whether the rep email field has any text in it.

**Review by tomas-nava — APPROVED** [review]

File: app/models/tasks/assign_hearing_disposition_task.rb [hunk 5]
```diff
@@ -184,7 +184,8 @@ def reschedule(
     scheduled_time_string:,
     hearing_location: nil,
     virtual_hearing_attributes: nil,
-    notes: nil
+    notes: nil,
+    email_recipients: nil
```
Comment (tomas-nava) [inline app/models/tasks/assign_hearing_disposition_task.rb:188]: I suggest you name this param consistent with `virtual_hearing_attributes` above
```ruby
    email_recipients_attributes: nil
```

**Review by tomas-nava — APPROVED** [review]

**Review by rubaiyat22 — APPROVED** [review]

**Pull request merged**